frontend/src/metabase/home/components/EmbedHomepage/tests/common.unit.spec.tsx

```diff
@@ -84,14 +89,120 @@ describe("EmbedHomepage (OSS)", () => {
 
     await userEvent.click(screen.getByText("Embedding done, all good"));
 
-    const lastCall = fetchMock.lastCall(
-      "path:/api/setting/embedding-homepage",
-      {
-        method: "PUT",
-      },
-    );
+    const lastCall = getLastHomepageSettingSettingCall();
 
     const body = await lastCall?.request?.json();
     expect(body).toEqual({ value: "dismissed-done" });
   });
+
+  it("should set 'embedding-homepage' to 'dismissed-not-interested-now' when dismissing because not interested", async () => {
+    setup();
+    await userEvent.hover(screen.getByText("Hide these"));
+
+    await userEvent.click(screen.getByText("I'm not interested right now"));
+
+    const lastCall = getLastHomepageSettingSettingCall();
+
+    const body = await lastCall?.request?.json();
+    expect(body).toEqual({ value: "dismissed-not-interested-now" });
+  });
+
+  describe("Feedback modal", () => {
+    const setupForFeedbackModal = async () => {
```

Comment:
Would it make more sense to put this in `beforeEach` under this describe instead? I see you're calling `setupForFeedbackModal` under every `it` inside this describe.

I thought it would make sense to have this function if you don't want to always call `setupForFeedbackModal` in every test.

Reply:
I think we usually don't do that, or do it very rarely.
It could make sense in this case but because 95% of the tests only clean mocks during beforeEach I'm not used to go and look for them

Reply:
well, that seems to vary. I'm used to putting setup steps for the same scenarios in `beforeEach`. e.g. if I have a few describes 
1. `describe("when condition A")`
2. `describe("when condition B")`
3. `describe("when condition C")`

I'd put `beforeEach` in each describe block with different setup steps.

At the end of the day, there's no convention for this I believe if I wouldn't mind you leave it as is as it seems this is still subjective.